### PR TITLE
COG-1269 FIX: enabled value should be set for user added to the index…

### PR DESCRIPTION
**Pull request by sorych**
… on create

### JIRA Issue Link
- [COG-1269](https://osi-cwds.atlassian.net/browse/COG-1269)

### Technical Description
 FIX: enabled value should be set for user added to the index on create

### Tests
- [ ] I have included unit tests 
- [x] I have included integration tests 
- [ ] I have included performance tests 
- [ ] I have included other tests 
- [ ] I have NOT included tests 

### Types of changes

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (No behavioral changes)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Checklist:

- [ ] My code is in a stable state ready to be deployable, but not necessarily complete.
- [x] I promise on my honor as a CWDS developer to ensure I don't break the build, to check Lint/Rubocop, use best practices, to leave the code in better shape than I found it, and to have a good day.

### CODE REVIEW

File: src/test/java/gov/ca/cwds/idm/service/UserServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
+package gov.ca.cwds.idm.service;
```
Comment (dmytro-palczewski-osi) [inline src/test/java/gov/ca/cwds/idm/service/UserServiceTest.java]: Could You test it by integration test? Each such standalone Spring test class adds about 10-20 secs to test execution.

Comment (kelfink) [reply src/test/java/gov/ca/cwds/idm/service/UserServiceTest.java]: It looks like you have included a unit test. And I agree an integration test would be nice.

Comment (sorych) [reply src/test/java/gov/ca/cwds/idm/service/UserServiceTest.java]: great catch, reworked it this way

File: src/test/java/gov/ca/cwds/idm/CreateUserTest.java [hunk 2]
```diff
@@ -552,4 +554,12 @@ private void testCreateUserValidationError(User user) throws Exception {
       this.invitationRequest = invitationRequest;
     }
   }
+
+  private class UserMatcher implements ArgumentMatcher<User> {
+
+    @Override
+    public boolean matches(User user) {
+      return Boolean.TRUE.equals(user.getEnabled()) && user.getStatus() != null;
```
Comment (dmytro-palczewski-osi) [inline src/test/java/gov/ca/cwds/idm/CreateUserTest.java]: equals.("FORCE_CHANGE_PASSWORD)

Comment (sorych) [reply src/test/java/gov/ca/cwds/idm/CreateUserTest.java]: why should I rely on what Cognito returns me?

Comment (sorych) [reply src/test/java/gov/ca/cwds/idm/CreateUserTest.java]: there is no requirement that status should be exactly this one

Comment (dmytro-palczewski-osi) [reply src/test/java/gov/ca/cwds/idm/CreateUserTest.java]: but you rely here:
Boolean.TRUE.equals(user.getEnabled())

Comment (dmytro-palczewski-osi) [reply src/test/java/gov/ca/cwds/idm/CreateUserTest.java]: "there is no requirement that status should be exactly this one"
there is a requirement to return ehat was returned by Cognito. And Cognito mock returns  "FORCE_CHANGE_PASSWORD".

Comment (sorych) [reply src/test/java/gov/ca/cwds/idm/CreateUserTest.java]: @dmytro-palczewski-osi  I refactored the whole approach, please check

File: src/test/java/gov/ca/cwds/idm/util/TestCognitoServiceFacade.java [hunk 3]
```diff
@@ -82,6 +82,8 @@
   public static final String SUPER_ADMIN_ID = "a3427f8e-b080-4f71-9514-840c1fcb505b";
 
   public static final String USERPOOL = "userpool";
+  public static final String COGNITO_USER_STATUS = "FORCE_CHANGE_PASSWORD";
```
Comment (dmytro-palczewski-osi) [inline src/test/java/gov/ca/cwds/idm/util/TestCognitoServiceFacade.java]: better NEW_COGNITO_USER_STATUS

Comment (sorych) [reply src/test/java/gov/ca/cwds/idm/util/TestCognitoServiceFacade.java]: what is NEW COGNITO? )

File: src/test/java/gov/ca/cwds/idm/util/TestCognitoServiceFacade.java [hunk 4]
```diff
@@ -82,6 +82,8 @@
...
+  public static final Boolean COGNITO_USER_ENABLED = Boolean.TRUE;
```
Comment (dmytro-palczewski-osi) [inline src/test/java/gov/ca/cwds/idm/util/TestCognitoServiceFacade.java]: NEW_COGNITO_USER_ENABLED_STATUS ?

Comment (sorych) [reply src/test/java/gov/ca/cwds/idm/util/TestCognitoServiceFacade.java]: done

File: src/test/java/gov/ca/cwds/idm/util/TestCognitoServiceFacade.java (hunk 3 above)
Comment (sorych) [reply src/test/java/gov/ca/cwds/idm/util/TestCognitoServiceFacade.java]: changed to COGNITO_USER_STATUS_ON_CREATE

**Pull request merged**